piv/key.go

```diff
@@ -709,6 +718,125 @@ func (yk *YubiKey) PrivateKey(slot Slot, public crypto.PublicKey, auth KeyAuth)
 	}
 }
 
+// SetPrivateKeyInsecure is an insecure method which imports a private key into the slot.
+// Users should almost always use GeneratePrivateKey() instead.
+//
+// Importing a private key breaks functionality provided by this package, including
+// AttestationCertificate() and Attest(). There are no stability guarantees for other
+// methods for imported private keys.
+//
+// Keys generated outside of the YubiKey should not be considered hardware-backed,
+// as there's no way to prove the key wasn't copied, exfiltrated, or replaced with malicious
+// material before being imported.
+func (yk *YubiKey) SetPrivateKeyInsecure(key [24]byte, slot Slot, private crypto.PrivateKey, policy Key) error {
+	params := make([][]byte, 0)
+
+	var paramTag byte
+	var elemLen int
+
+	switch priv := private.(type) {
+	case *rsa.PrivateKey:
+		paramTag = 0x01
+		switch priv.N.BitLen() {
+		case 1024:
+			policy.Algorithm = AlgorithmRSA1024
+			elemLen = 64
+		case 2048:
+			policy.Algorithm = AlgorithmRSA2048
+			elemLen = 128
+		default:
+			return errUnsupportedKeySize
+		}
+
+		// P value
+		params = append(params, priv.Primes[0].Bytes())
+
+		// Q value
+		params = append(params, priv.Primes[1].Bytes())
+
+		// dP value
+		params = append(params, priv.Precomputed.Dp.Bytes())
+
+		// dQ value
+		params = append(params, priv.Precomputed.Dq.Bytes())
+
+		// Qinv value
+		params = append(params, priv.Precomputed.Qinv.Bytes())
+	case *ecdsa.PrivateKey:
+		paramTag = 0x6
+		size := priv.PublicKey.Params().BitSize
+		switch size {
+		case 256:
+			policy.Algorithm = AlgorithmEC256
+			elemLen = 32
+		case 384:
+			policy.Algorithm = AlgorithmEC384
+			elemLen = 48
+		default:
+			return fmt.Errorf("unsupported curve: %d", size)
+		}
+
+		// S value
+		privateKey := make([]byte, elemLen)
+		params = append(params, priv.D.FillBytes(privateKey))
```

Comment:
nit: FillBytes is Go 1.15+ but we probably still want to support Go 1.14 until Go 1.16 is out. Can you please replace this with a different method?

https://golang.org/pkg/math/big/#Int.FillBytes
https://golang.org/doc/devel/release.html#policy